### DISCUSSION TITLE
Ios game made with Animate/Air. App Icon doesn't show when app is minimized?

### DISCUSSION THREAD

**Discussion by FrancescoMuja** [airsdk/Adobe-Runtime-Support#1903] [Q&A]
Hello everyone!
I created a video game with Animate and the Air SDK, and it is now ready to be published on the app store!
Everything is fine, except for this problem: when I test the app (ad hoc profile) on my iPad (9th generation, ios 15.4), and press the home button twice to minimize it, as you can see in the following photo, the game icon does not appear!
All the other icons work, that's the only one that doesn't.
What could be the problem?
![Senza-nome-1](https://user-images.githubusercontent.com/105378305/167899657-3c54f9e8-dbb2-4529-8d3f-a325283c348c.png)

I created assets.car and LaunchScreen files with Xcode following this [guide](https://airnativeextensions.github.io/tutorials/ios-icons-assets-car.html#method-1-command-line), then I also included all the icons required in Air's publish settings and, as I said before, they all seem to work fine, including Harman's launch screen.
I was also able to upload the build to Apple Connect with no error messages.
Why is that icon the only one that doesn't appear? I reckon it's the 60x60 icon? It might be a minor thing, but it's bugging me!
What did I do wrong?
Thanks in advance!

**Comment by FrancescoMuja** [comment]
UPDATE: I thought it might be worth to mention a few other things.

That icon doesn't work on iPhones, either. At least, on the one I used to test it again (iPhone 5s)
Just like on my iPad:
![screenshot-iphone](https://user-images.githubusercontent.com/105378305/168024240-95fda5cb-093f-4aa0-8135-39b2189f57c7.png)

On the other hand, on my android phone (Redmi Note 8 Pro) it works perfectly.
But then again, the Play Store doesn't try to complicate even the easiest things...
![screenshot-android](https://user-images.githubusercontent.com/105378305/168024592-600a7ac9-c494-4a23-bd13-7037ff28c541.png)

I should probably add that I'm installing the ad-hoc build on my ios devices via testapp.io, don't know if that could have anything to do with it. My next step will be to test the app store build with TestFlight.

**Reply by marchbold** [reply]
Have you double checked that asset is in the assets catalogue (Assets.car)?

**Reply by marchbold** [reply]
Also this is the latest version of that tutorial: https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car

**Reply by FrancescoMuja** [reply]
Hello Michael! Thank you for replying!
> Have you double checked that asset is in the assets catalogue (Assets.car)?

They are on Xcode, but once assets.car is created I don't know how to "open" it to see its contents and doublecheck.

But I've recently noticed something else. 
After I "build the product" on Xcode, inside the .app package there are these two other icons which are outside the Assets.car catalog:
AppIcon60x60@2x.png
AppIcon76x76@2x-ipad.png
<img width="310" alt="Schermata 2022-05-12 alle 16 34 46" src="https://user-images.githubusercontent.com/105378305/168102092-ea883b86-27de-46aa-9f51-143df539c4a7.png">

Could it be that's the problem? How could I integrate those? I tried to add them to the Air package, but it didn't do much. 

Another question... Do I have to add any line of code in the .xml descriptor to integrate the assets.car?

**Reply by marchbold** [reply]
That looks odd with the image there. Have you correctly assigned those 2 assets to particular sizes in your app icon inside the assets catalogue? 

> Do I have to add any line of code in the .xml descriptor to integrate the assets.car?

No the Assets.car should be picked up automatically as long as it's in the root of your package.

**Reply by FrancescoMuja** [reply]
Hello again, Michael!

>No the Assets.car should be picked up automatically as long as it's in the root of your package.

Good, thanks! That means I didn't skip a passage.

So, I've found a way to open the assets.car catalogue and examine its content, with the AssetCatalogTinkerer app.
Here is how it looks like:
![screenshot-assetscar](https://user-images.githubusercontent.com/105378305/168281889-f5913b02-3836-48f8-ade0-48e0d525e6d8.png)
I've circled the 60x60 and 76x76 icons to highlight that they are indeed inside the catalogue.
And this is how I set it up on Xcode... Each icon is in place, there are no error messages, they're all in a png format, I just can't understand what I'm doing wrong
![screenshot-xcode](https://user-images.githubusercontent.com/105378305/168281929-19698fd7-c008-4d11-b33a-e848dbc7a144.png)
And I don't know what those two icons outside the assets.car are for, or if they're even part of the problem.

I'm stumped..!

**Reply by ajwfrost** [reply]
For the two icons outside of assets.car, these may be ones that are listed in your application descriptor file as icons e.g. 
```
<icon>
	<image120x120>icons/icon_120.png</image120x120>
</icon>
```
and I think if you have that sort of line, it will generate the 60@2x image.

Do you have one that's using `image60x60`? i.e. a 60x60 pixel image listed in your icon set?

The way Apple set up the assets.car mechanism, I think means that as well as these icons being included in the .car file, they also need to be mentioned by the Info.plist file, and this is then generated from the app descriptor file. So, worth playing with the values in there for the resolutions that are going wrong...

thanks

**Reply by FrancescoMuja** [reply]
Hey Michael and Andrew, 
><image120x120>icons/icon_120.png</image120x120>

Yes, I have that line in the descriptor. Actually, I have a line like that for ALL the sizes required in the Air publishing settings, just to be on the safe side.

But I finally solved the issue: I just remade all the icons from scratch and tested the app on Testflight and voila, it works perfectly.

I'm quite ashamed to admit that I would have saved your precious time if I had done it before.
I'm sorry :(
But I'm really grateful for your help, thanks again.

**Reply by marchbold** [reply]
Sometimes when testing iOS cache's the asset catalogue between installs so it's actually really hard to test app icon and launch screen changes. Glad you got it working!

**Reply by FrancescoMuja** [reply]
> Sometimes when testing iOS cache's the asset catalogue between installs so it's actually really hard to test app icon and launch screen changes. Glad you got it working!

That's something I have to keep in mind. Thank you!
